File: server/Models/Ingredient.cs
using System.ComponentModel.DataAnnotations;

namespace allspice.Models;

public class Ingredient : RepoItem<int>
{
  [MaxLength(255)] public string Name { get; set; }
  [MaxLength(255)] public string Quantity { get; set; }
  public int RecipeId { get; set; }
}

File: client/src/models/Ingredient.js
import { DatabaseItem } from "./DatabaseItem.js";

/**
   * @typedef {Object} IngredientData
   * @property {number} id
   * @property {Date} createdAt
   * @property {Date} updatedAt
   * @property {string} name
   * @property {string} quantity
   * @property {number} recipeId
   * @param {IngredientData} data 
   */

export class Ingredient extends DatabaseItem{
  constructor(data){
    super(data)
    this.name = data.name
    this.quantity = data.quantity
    this.originRecipeId = data.recipeId
  }
}